### PR TITLE
completed react-file-uploads

### PR DESCRIPTION
**Pull request by do-jonathan4**
image
F1

### CODE REVIEW

File: react-file-uploads/server/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  const { caption } = req.body;
+  if (!caption) {
+    throw new ClientError(400, 'caption is a required field');
+  }
+  /**
+   * - create a url for the image by combining '/images' with req.file.filename
+   * - insert the "caption" and "url" into the "images" table
+   * - respond with the inserted row data
+   * - catch any errors
+   */
```
Comment (scbowler) [inline react-file-uploads/server/index.js:44]: Looks like you missed the server side portion of the exercise.

**Review by scbowler — CHANGES_REQUESTED** [review]
The React, client side, looks good. However looks like you missed the server side of the exercise.

**Comment by do-jonathan4** [comment]
image

File: react-file-uploads/server/index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  const sql = `
+    insert into "images" ("caption", "url")
+    values ($1, $2)
+    returning "caption", "createdAt", "imageId", "url"
+  `;
+  const params = [caption, imageUrl];
+  db.query(sql, params)
+    .then(result => {
+      const [user] = result.rows;
+      res.status(201).json(user);
```
Comment (scbowler) [inline react-file-uploads/server/index.js:36]: Watch your variable names, this isn't a `user`. Something like `image` or something would be more appropriate.

**Review by scbowler — APPROVED** [review]
Looks good 👍 

One FYI comment below, no changes required.

**Pull request merged**